static/js/apps/nl_interface/data_app/app.tsx

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Main component for the data version of the NL interface.
+ */
+
+import React, { useEffect, useRef, useState } from "react";
+
+import { getUrlToken } from "../../../tools/stat_var/util";
+import { QueryHistory } from "../query_history";
+import { QueryResult } from "../query_result";
+import { QuerySearch } from "../query_search";
+
+const NEXT_PROMPT_DELAY = 5000;
+
+export function App(): JSX.Element {
+  const [queries, setQueries] = useState<string[]>([]);
+  const [contextList, setContextList] = useState<any[]>([]);
+  const autoRun = useRef(!!getUrlToken("a"));
+  const urlPrompts = useRef(getUrlPrompts());
+  // Timer used to wait NEXT_PROMPT_DELAY ms before inputting a new prompt.
+  const nextPromptDelayTimer = useRef(null);
```

Comment:
no need for the timer?

Reply:
this one I wasn't sure. If there are multiple queries, do we want to delay between each prompt? or no?

Reply:
If there are multiple queries, the caller will drive it, so assume there's a single query?

Reply:
I feel the data endpoint should only do single query (hence can remove all the logic to multiple queries)

Reply:
done